apps/contact-service/src/routes/clubs/member/getClubContacts.ts

```diff
@@ -0,0 +1,57 @@
+import {NotFoundError} from '@vexl-next/domain/src/general/commonErrors'
+import {
+  GetClubContactsErrors,
+  UserNotClubMemberError,
+} from '@vexl-next/rest-api/src/services/contact/contracts'
+import {GetClubContactsEndpoint} from '@vexl-next/rest-api/src/services/contact/specification'
+import makeEndpointEffect from '@vexl-next/server-utils/src/makeEndpointEffect'
+import {validateChallengeInBody} from '@vexl-next/server-utils/src/services/challenge/utils/validateChallengeInBody'
+import {Effect} from 'effect'
+import {Handler} from 'effect-http'
+import {ClubMembersDbService} from '../../../db/ClubMemberDbService'
+import {ClubsDbService} from '../../../db/ClubsDbService'
+
+export const getClubContacts = Handler.make(GetClubContactsEndpoint, (req, _) =>
+  makeEndpointEffect(
+    Effect.gen(function* (_) {
+      yield* _(validateChallengeInBody(req.body))
+
+      const clubsDb = yield* _(ClubsDbService)
+      const clubMembersDb = yield* _(ClubMembersDbService)
+
+      const club = yield* _(
+        clubsDb.findClubByUuid({uuid: req.body.clubUuid}),
+        Effect.flatten,
+        Effect.catchTag('NoSuchElementException', (e) => new NotFoundError())
+      )
+
+      const memberData = yield* _(
+        clubMembersDb.findClubMemberByPublicKey({
+          publicKey: req.body.publicKey,
+        }),
+        Effect.flatten,
+        Effect.catchTag('NoSuchElementException', (e) => new NotFoundError())
+      )
+
+      if (club.id !== memberData.clubId) {
+        return yield* _(new UserNotClubMemberError())
```

Comment:
See commend above. I would remove this error all together. I think it's better to return 404 if user is not a member and not tell user the club exists...